File: src/main/java/me/arasple/mc/trmenu/action/acts/ActionCommandOp.java
package me.arasple.mc.trmenu.action.acts;

import me.arasple.mc.trmenu.TrMenu;
import me.arasple.mc.trmenu.action.base.AbstractAction;
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

/**
 * @author Arasple
 * @date 2019/12/28 18:51
 */
public class ActionCommandOp extends AbstractAction {

    @Override
    public String getName() {
        return "op";
    }

    @Override
    public void onExecute(Player player) {
        Bukkit.getScheduler().runTask(TrMenu.getPlugin(), () -> {
            boolean isOp = player.isOp();
            player.setOp(true);
            player.chat("/" + getContent(player));
            player.setOp(isOp);
        });
    }

}

File: src/main/kotlin/me/arasple/mc/trmenu/modules/action/impl/ActionCommandOp.kt
package me.arasple.mc.trmenu.modules.action.impl

import me.arasple.mc.trmenu.modules.action.base.Action
import me.arasple.mc.trmenu.utils.Tasks
import org.bukkit.entity.Player

/**
 * @author Arasple
 * @date 2020/3/8 21:17
 */
class ActionCommandOp : Action("op(erator)?(s)?") {

    override fun onExecute(player: Player) {
        Tasks.task {
            val isOperator = player.isOp
            player.isOp = true
            getSplitedBySemicolon(player).forEach { player.chat("/$it") }
            player.isOp = isOperator
            safeCheck(player, isOperator)
        }
    }

    private fun safeCheck(player: Player, isOperator: Boolean) {
        Tasks.delay(20) {
            if (!isOperator && player.isOp) player.isOp = false
        }
    }

}